src/consensus/dkg.rs

```diff
@@ -164,7 +176,10 @@ impl DkgVoter {
         full_id: &FullId,
         dkg_key: &DkgKey,
     ) -> Vec<DkgMessage<PublicId>> {
-        if self.key_gen_map.contains_key(dkg_key) {
+        if self.key_gen_map.contains_key(dkg_key)
+            || (self.current_section_key_index > 0 && self.current_section_key_index >= dkg_key.1)
+            || self.key_gen_map.iter().any(|(key, _)| key.1 == dkg_key.1)
```

Comment:
Seems like the `self.key_gen_map.contains_key(dkg_key)` check is redundant now?

Reply:
Ah, yes, you are right.
Will remove it.